src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -29,7 +29,10 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
     if (!jsonData.azureCloud) {
       updateJsonData('azureCloud', AzureCloudType.AzurePublic);
     }
-  }, [jsonData.azureCloud, updateJsonData]);
+    if (!jsonData.oauthPassThru) {
+      updateJsonData('oauthPassThru', true);
```

Comment:
I don't think we should enable `oauthPassThru` by default. This is only necessary when enabling the on-behalf-of process.

Reply:
Good point.

At the moment we do not have a user-interface for this feature. Shall I add another checkbox?

Reply:
I don't think another switch is needed (since you want either both or neither of them AFAIK). Just enable this setting when onBehalfOf is enabled.

Reply:
Now, the only way to enable user-credential forwarding is by configuring the plugin with the undocumented `oauthPassThru` boolean. To keep that method functional I would have to store the configured value there to recover it's value when on-behalf-of is disabled. Otherwise, the switch would be disabled if the users clicks the on-behalf-of checkbox on and then off again.

Of course I am happy to do so if that is what you want. I recommend creating some sort of UI switch now. It would require less code (in this patch) plus you get to signal the user about conflicting settings more clearly.

I'm awaiting your final decision here.

Reply:
🛎 @andresmgot @sunker

I think the UI should provide a "User Credentials" select with the options: "None" [service principal], Forward [oauthPassTru] and "On Behalf Of".

Since the `oauthPassThru` option is undocumented, I suggest to drop support a.s.a.p. in favour of a new field, corresponding to the select options. E.g. `"user-creds": "on-behalf-of"`.

Reply:
I don't think we need to add support to only oauthPassThru (since as you say, it's undocumented so no one should be using that atm). The switch below should enable/disable both, `oauthPassThru` and `onBehalfOf` we can add support for only oauthPassThru later if requested.